servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -3760,6 +3776,12 @@ Policy getPolicy(ObjectStoreConnection con, String domainName, String policyName
         if (policy != null) {
             policy.setAssertions(con.listAssertions(domainName, policyName, version));
         }
+
+        Map<String, TagValueList> policyTags = con.getPolicyTags(domainName, policyName, version);
```

Comment:
there is no reason to get the policyTags if the policy is null. so we should have:

if (policy != null) {
   Map<String, TagValueList> policyTags = con.getPolicyTags(domainName, policyName, version);
...

and not check if policy != null after we get the tags

Reply:
this was not addressed - this is a wasted DB call if the policy is null. use the if (policy != null) block from 3776 to avoid unnecessary calls if policy is null